File: Daily Questions/2220-minimum-bit-flips-to-convert-number/2220-minimum-bit-flips-to-convert-number.c
#include <stdio.h>

/**
 * Function to calculate the minimum number of bit flips required to
 * convert `start` to `goal`.
 *
 * @param start (int): The initial integer.
 * @param goal (int): The target integer.
 * @return int: The number of bit flips required.
 */
int minBitFlips(int start, int goal) {
    int c = 0;

    // Continue until both 'start' and 'goal' become 0
    while (start > 0 || goal > 0) {
        // Compare the least significant bits using bitwise AND operation
        if ((start & 1) != (goal & 1)) {
            c++;  // Increment the counter if the bits are different
        }
        
        // Right shift both 'start' and 'goal' by 1 bit
        start >>= 1;
        goal >>= 1;
    }
    
    return c;
}

File: Daily Questions/2220-minimum-bit-flips-to-convert-number/2220-minimum-bit-flips-to-convert-number.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        """
        This function calculates the minimum number of bit flips required
        to convert 'start' to 'goal'. A bit flip means changing a '0' to '1' 
        or vice versa.

        The XOR (^) operation between 'start' and 'goal' will produce a binary 
        number where each '1' indicates a differing bit between 'start' and 'goal'.
        The number of '1's in the XOR result is the number of bit flips required.
        
        Parameters:
        start (int): The initial integer.
        goal (int): The target integer.

        Returns:
        int: The number of bit flips required.
        """
        # XOR 'start' and 'goal' to get a binary number where each '1' represents a differing bit.
        # Convert the XOR result to a binary string using bin(), and count the number of '1's 
        # which represent the differing bits.
        return bin(start ^ goal).count('1')
